### PR TITLE
add example code for public APIs

### PR DESCRIPTION
**Pull request by HydrogenSulfate**
### PR types

Others
### PR changes

Docs
### Describe

主要修改点：
1. 为公开API的docstring添加example code

### CODE REVIEW

File: ppsci/loss/l1.py [hunk 1]
```diff
@@ -28,6 +28,10 @@ class L1Loss(base.LossBase):
     \end{cases}
     $$
 
+    Examples:
+        ``` python
+        >>> loss = ppsci.geometry.L1Loss("mean")
```
Comment (zhiminzhang0830) [inline ppsci/loss/l1.py:33]: 调用路径错误

File: ppsci/loss/l2.py [hunk 2]
```diff
@@ -23,6 +23,11 @@ class L2Loss(base.LossBase):
     $$
     L = \sum\limits_{i=1}^{N}{(x_i-y_i)^2}
     $$
+
+    Examples:
+        ``` python
+        >>> loss = ppsci.geometry.L2Loss()
```
Comment (zhiminzhang0830) [inline ppsci/loss/l2.py:29]: 同上

File: ppsci/loss/mse.py [hunk 3]
```diff
@@ -33,6 +33,10 @@ class MSELoss(base.LossBase):
     \end{cases}
     $$
 
+    Examples:
+        ``` python
+        >>> loss = ppsci.geometry.MSELoss("mean")
```
Comment (zhiminzhang0830) [inline ppsci/loss/mse.py:38]: 同上

File: ppsci/optimizer/lr_scheduler.py [hunk 4]
```diff
@@ -562,6 +586,10 @@ def step(self, epoch=None):
 class CosineWarmRestarts(LRBase):
     """Set the learning rate using a cosine annealing schedule with warm restarts.
 
+    Examples:
+        ``` python
+        >>> lr = ppsci.optimizer.lr_scheduler.CosineWarmRestarts(10, 1, 1e-3, 0, 0.5)
```
Comment (zhiminzhang0830) [inline ppsci/optimizer/lr_scheduler.py:591]: 参数T_0必须为正数，T_mult必须大于1，可修改为lr = ppsci.optimizer.lr_scheduler.CosineWarmRestarts(10, 1, 1e-3, 14, 2)

File: ppsci/optimizer/optimizer.py [hunk 5]
```diff
@@ -30,6 +30,11 @@
 class SGD:
     """Stochastic Gradient Descent.
 
+    Examples:
+        ``` python
+        >>> model = ppsci.arch.MLP(("x",), ("u"), 5, 20)
```
Comment (zhiminzhang0830) [inline ppsci/optimizer/optimizer.py:35]: 修改为model = ppsci.arch.MLP(("x",), ("u",), 5, 20)

File: ppsci/optimizer/optimizer.py [hunk 6]
```diff
@@ -212,6 +227,11 @@ class LBFGS:
     """The L-BFGS is a quasi-Newton method for solving an unconstrained optimization
         problem over a differentiable function. Closely related is the Newton method for minimization.
 
+    Examples:
+        ``` python
+        >>> model = ppsci.arch.MLP(("x",), ("u"), 5, 20)
```
Comment (zhiminzhang0830) [inline ppsci/optimizer/optimizer.py:232]: 同上

File: ppsci/optimizer/optimizer.py [hunk 7]
```diff
@@ -320,6 +345,11 @@ def __call__(self, model_list: Tuple[nn.Layer, ...]):
 class AdamW:
     """AdamW is implemented based on DECOUPLED WEIGHT DECAY REGULARIZATION.
 
+    Examples:
+        ``` python
+        >>> model = ppsci.arch.MLP(("x",), ("u"), 5, 20)
```
Comment (zhiminzhang0830) [inline ppsci/optimizer/optimizer.py:350]: 同上

File: ppsci/solver/solver.py [hunk 8]
```diff
@@ -41,6 +41,31 @@
...
+        ...     {"u": lambda out: out["u"]},
+        ...     {"u": 0},
+        ...     rect,
+        ...     {
+        ...         "dataset": "IterableNamedArrayDataset",
+        ...         "iters_per_epoch": 1,
+        ...         "batch_size": 16,
+        ...     },
+        ...     ppsci.loss.MSELoss("mean"),
+        ...     name="BC",
```
Comment (zhiminzhang0830) [inline ppsci/solver/solver.py:59]: name不应为“BC”

File: ppsci/solver/solver.py [hunk 9]
```diff
@@ -41,6 +41,31 @@
...
+        ... )
+        >>> solver = Solver(
```
Comment (zhiminzhang0830) [inline ppsci/solver/solver.py:61]: 是否需要修改为ppsci.solver.Solver

File: ppsci/utils/download.py [hunk 10]
```diff
@@ -53,13 +53,12 @@ def get_weights_path_from_url(url, md5sum=None):
...
 
     Examples:
-        .. code-block:: python
-
-            from paddle.utils.download import get_weights_path_from_url
-
-            resnet18_pretrained_weight_url = "https://paddle-hapi.bj.bcebos.com/models/resnet18.pdparams"
-            local_weight_path = get_weights_path_from_url(resnet18_pretrained_weight_url)
+        ``` python
+        >>> from paddle.utils.download import get_weights_path_from_url
```
Comment (zhiminzhang0830) [inline ppsci/utils/download.py:57]: from ppsci.utils.download import get_weights_path_from_url

File: ppsci/visualize/visualizer.py [hunk 11]
```diff
@@ -27,6 +27,21 @@
...
+        ...         ("t_f",),
+        ...         {
+        ...             "eta": lambda d: d["eta"],  # plot with latex title
+        ...             "eta_gt": lambda d: d["eta_gt"],  # plot with latex title
+        ...         },
+        ...         num_timestamps=1,
+        ...         prefix="viv_pred",
+        ...     )
+        ... }
+        ```
```
Comment (zhiminzhang0830) [inline ppsci/visualize/visualizer.py:44]: 这个类的参数batchsize的docstring是否应该是batch_size (int, optional)而不是batch_size (int)？

File: ppsci/visualize/visualizer.py [hunk 12]
```diff
@@ -57,6 +72,20 @@ def save(self, filename, data_dict):
 class VisualizerScatter3D(base.Visualizer):
     """Visualizer for 3d scatter data.
 
+    Examples:
+        ``` python
+        >>> visualizer = {
+        ...     "visulzie_states": ppsci.visualize.VisualizerScatter3D(
+        ...         vis_datas,
+        ...         {
+        ...             "pred_states": lambda d: output_transform(d),
```
Comment (zhiminzhang0830) [inline ppsci/visualize/visualizer.py:81]: output_transforme是否需要给出定义？或者修改为`"pred_states": lambda d: d["pred_states"],`？

File: ppsci/visualize/visualizer.py [hunk 13]
```diff
@@ -101,6 +130,17 @@ def save(self, filename, data_dict):
 class VisualizerVtu(base.Visualizer):
     """Visualizer for 2D points data.
 
+    Examples:
+        ``` python
+        >>> visualizer = {
+        ...     "visulzie_u_v": ppsci.visualize.VisualizerVtu(
+        ...         vis_points,
+        ...         {"u": lambda d: d["u"], "v": lambda d: d["v"], "p": lambda d: d["p"]},
+        ...         num_timestamps=ntime_all,
```
Comment (zhiminzhang0830) [inline ppsci/visualize/visualizer.py:139]: vis_points、ntime_all未预先定义

**Review by zhiminzhang0830 — APPROVED** [review]
